src/path-decomposition.h

```diff
@@ -0,0 +1,81 @@
+#include <vector>
+#include <string>
+#include <boost/graph/adjacency_list.hpp>
+#include "../tdlib/src/treedec_traits.hpp"
+
+typedef boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS> Graph;
+typedef boost::graph_traits<Graph>::vertex_descriptor vertex_t;
+typedef treedec::graph_traits<Graph>::treedec_type Tree;
+typedef boost::graph_traits<Tree>::vertex_descriptor tree_vertex;
+
+class PathDecomposition {
+    std::vector<std::vector<vertex_t>> _bags;
+    Graph _g;
+
+public:
+
+    enum error_types {
+        EDGES = 0,
+        NEGATIVE_VERTICES = 1,
+        INCONTINUITY = 2,
+        VERTICES = 3
+    };
+
+    struct CorectnessException : public std::exception{
+        /*
+        Thrown when path-width decomposition is incorrect
+        */
+    private:
+    public:
+        int _error_type;
+        int _bag_i, _bag_j, _bag_k, _violating_node;
+
+        CorectnessException(int error_type);
+        /*
+        @error_type == 0 used when in graph |_g| exists edge (v, u) but in |_bags| there is no bag that contains {v, u}
+        @error_type == 1 used when in |_bags| exists vertex |v| that does not belong to [0; |V(_g)|)
+        @error_type == 3 used when exists vertex |v| that does not belong to |_bags|
+        */
+        CorectnessException(int i, int j, int k, int u);
```

Comment:
what is u? i -> bag_i (for j and k as well)

Reply:
There is a comment below

Reply:
There is no u in the comment. Please give the arguments more meaningful names.